CONTRIBUTING.md

```diff
@@ -0,0 +1,173 @@
+
+Good pull requests - patches, improvements, new features - are a fantastic
+help. They should remain focused in scope and avoid containing unrelated
+commits.
+
+**Please ask first** before embarking on any significant pull request (e.g.
+implementing features, refactoring code, porting to a different language),
+otherwise you risk spending a lot of time working on something that the
+project's developers might not want to merge into the project.
+
+Please adhere to the coding conventions used throughout a project (indentation,
+accurate comments, etc.) and any other requirements (such as test coverage).
+
+Follow this process if you'd like your work considered for inclusion in the
+project:
+
+1. [Fork](http://help.github.com/fork-a-repo/) the project, clone your fork,
+   and configure the remotes:
+
+   ```bash
+   # Clone your fork of the repo into the current directory
+   git clone git@github.com:WeTransfer/UINotifications.git
+   # Navigate to the newly cloned directory
+   cd UINotifications
+   # Assign the original repo to a remote called "upstream"
+   git remote add upstream git@github.com:WeTransfer/UINotifications.git
+   ```
+
+2. If you cloned a while ago, get the latest changes from upstream:
+
+   ```bash
+   git checkout <dev-branch>
+   git pull upstream <dev-branch>
+   ```
+
+3. Create a new topic branch (off the main project development branch) to
+   contain your feature, change, or fix:
+
+   ```bash
+   git checkout -b <topic-branch-name>
+   ```
+
+4. Commit your changes in logical chunks.
+
+5. Locally merge (or rebase) the upstream development branch into your topic branch:
+
+   ```bash
+   git pull [--rebase] upstream <dev-branch>
+   ```
+
+6. Push your topic branch up to your fork:
+
+   ```bash
+   git push origin <topic-branch-name>
+   ```
+
+7. [Open a Pull Request](https://help.github.com/articles/using-pull-requests/)
+    with a clear title and description.
+
+### Conventions of commit messages
+
+Adding features on repo
+
+```bash
+git commit -m "feat: message about this feature"
+```
+
+Fixing features on repo
+
+```bash
+git commit -m "fix: message about this update"
+```
+
+Removing features on repo
+
+```bash
+git commit -m "refactor: message about this" -m "BREAKING CHANGE: message about the breaking change"
+```
+
+
+**IMPORTANT**: By submitting a patch, you agree to allow the project owner to
+license your work under the same license as that used by the project.
+
+### Discussions
+
+We aim to keep all project discussion inside GitHub issues. This is to make sure valuable discussion is accessible via search. If you have questions about how to use the library, or how the project is running - GitHub issues are the goto tool for this project.
+
+## Ownership
+
+If you get a merged Pull Request, regardless of content (typos, code, doc fixes), then you're eligible for push access to this repository. This is checked for on pull request merges and an invite to the repository is sent via GitHub.
```

Comment:
Talked about it with @AvdLee, and it feels too strong to us. I think that this should be on a case by case basis (per repo and per contributors).
For example, the assumptions that are made on the next paragraph are not applicable to `WeScan`. We are moving to only having a `master` branch and no `develop` branch - that means that even with push access, contributors couldn't merge PRs.

Reply:
That's a fair point. I'll remove the section. Can you document the branch stuff in here when it's "rolled out"?

Reply:
Sounds good!